src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -26,73 +70,812 @@ const ActorInit En_Fz_InitVars = {
+    Vec3f pos; 
+
+    iceSmoke = this->iceSmoke;
+
+    for (i = 0; i < 40; i++) {
+        if (iceSmoke->freezingState) {
+            iceSmoke->pos.x += iceSmoke->velocity.x;
+            iceSmoke->pos.y += iceSmoke->velocity.y;
+            iceSmoke->pos.z += iceSmoke->velocity.z;
+            iceSmoke->numFramesActive++;
+            iceSmoke->velocity.x += iceSmoke->accel.x;
+            iceSmoke->velocity.y += iceSmoke->accel.y;
+            iceSmoke->velocity.z += iceSmoke->accel.z;
+
+            // Not Freezing
+            if (iceSmoke->freezingState == 1) {
+
+                // Becoming more opaque
+                if (iceSmoke->primAlphaState == 0) {
+                    iceSmoke->primAlpha += 10;
+                    if (iceSmoke->primAlpha >= 100) {
+                        iceSmoke->primAlphaState++;
+                    }
+                } else { // Becoming more transparent
+                    iceSmoke->primAlpha -= 3;
+                    if (iceSmoke->primAlpha <= 0) {
+                        iceSmoke->primAlpha = 0;
+                        iceSmoke->freezingState = 0;
+                    }
+                }
+            } else if (iceSmoke->freezingState == 2) { // Freezing
+                Math_SmoothScaleMaxF(&iceSmoke->xyScale, iceSmoke->xyScaleTarget, 0.1f,
+                                     iceSmoke->xyScaleTarget / 10.0f);
+
+                // Becoming more opaque
+                if (iceSmoke->primAlphaState == 0) {
+                    if (iceSmoke->numFramesActive >= 7) {
+                        iceSmoke->primAlphaState++;
+                    }
+                } else { // Becoming more transparent, slows down
+                    iceSmoke->accel.y = 2.0f;
+                    iceSmoke->primAlpha -= 17;
+                    iceSmoke->velocity.x *= 0.75f;
+                    iceSmoke->velocity.z *= 0.75f;
+                    if (iceSmoke->primAlpha <= 0) {
+                        iceSmoke->primAlpha = 0;
+                        iceSmoke->freezingState = 0;
+                    }
+                }
+
+                if ((this->unusedTimer2 == 0) && (iceSmoke->primAlpha >= 101) && iceSmoke->isTimerMod8) {
+                    this->collider3.dim.pos.x = (s16)iceSmoke->pos.x;
+                    this->collider3.dim.pos.y = (s16)iceSmoke->pos.y;
+                    this->collider3.dim.pos.z = (s16)iceSmoke->pos.z;
+                    CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider3);
+                }
+                pos.x = iceSmoke->pos.x;
+                pos.y = iceSmoke->pos.y + 10.0f;
+                pos.z = iceSmoke->pos.z;
+
+                // Stop if collision
+                if ((iceSmoke->primAlphaState != 2) && (EnFz_IsCollision(this, &pos))) {
+                    iceSmoke->primAlphaState = 2;
+                    iceSmoke->velocity.x = 0.0f;
+                    iceSmoke->velocity.z = 0.0f;
+                }
+            }
+        }
+        iceSmoke += 1;
```

Comment:
```suggestion
        iceSmoke ++;
```

Reply:
No clue how I missed that

Reply:
I also switched to ARRAY_COUNT(this->iceSmoke) for the for-loops